src/map.cpp

```diff
@@ -6999,6 +7001,173 @@ void map::rotten_item_spawn( const item &item, const tripoint &pnt )
     }
 }
 
+void map::actualize_snow( float new_snow_lvl )
+{
+    const int zmin = 0;
+    const int zmax = zlevels ? OVERMAP_HEIGHT : 0;
+    tripoint gp;
+    int &gx = gp.x;
+    int &gy = gp.y;
+    int &gz = gp.z;
+    for( gz = zmin; gz <= zmax; gz++ ) {
+        for( gx = 0; gx < my_MAPSIZE; gx++ ) {
+            for( gy = 0; gy < my_MAPSIZE; gy++ ) {
+                actualize_snow_in_submap( tripoint( gx, gy, gz ), new_snow_lvl, false );
```

Comment:
You can directly pass `gp` here:
```suggestion
                actualize_snow_in_submap( gp, new_snow_lvl, false );
```